group-coordinator/src/main/java/org/apache/kafka/coordinator/group/streams/topics/InternalTopicManager.java

```diff
@@ -255,9 +255,11 @@ private static CreatableTopic toCreatableTopic(final ConfiguredInternalTopic con
     }
 
     private static ConfiguredSubtopology fromPersistedSubtopology(final StreamsGroupTopologyValue.Subtopology subtopology,
+                                                                  final Map<String, TopicMetadata> topicMetadata,
                                                                   final Map<String, Integer> decidedPartitionCountsForInternalTopics
     ) {
         return new ConfiguredSubtopology(
+            computeNumberOfTasks(subtopology, topicMetadata, decidedPartitionCountsForInternalTopics),
```

Comment:
This is essentially just precomputing what can be derived from the topology and the topic metadata. It is meant to speed up accesses to the number of tasks during request validation and assignment.